Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Doctrine\Bundle\PHPCRBundle\Form\ChoiceList;
+
+use Symfony\Bridge\Doctrine\Form\ChoiceList\DoctrineChoiceLoader as BaseDoctrineChoiceLoader;
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\ODM\PHPCR\Mapping\ClassMetadata as PHPCRClassMetadata;
+use Doctrine\Common\Persistence\Mapping\ClassMetadata;
+use PHPCR\Util\UUIDHelper;
+use Symfony\Bridge\Doctrine\Form\ChoiceList\EntityLoaderInterface;
+use Symfony\Bridge\Doctrine\Form\ChoiceList\IdReader;
+use Symfony\Component\Form\ChoiceList\ChoiceListInterface;
+use Symfony\Component\Form\ChoiceList\Factory\ChoiceListFactoryInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+
+/**
+ * Supports UUIDs as choice values, it will automatically detect them and if the PHPCR document
+ * has a UUID field mapping it will be enabled.
+ *
+ * @author Steffen Brem <steffenbrem@gmail.com>
+ */
+class DoctrineChoiceLoader extends BaseDoctrineChoiceLoader
+{
+    /**
+     * @var ObjectManager
+     */
+    private $manager;
+
+    /**
+     * @var ClassMetadata
+     */
+    private $classMetadata;
+
+    /**
+     * @var IdReader
+     */
+    private $idReader;
+
+    /**
+     * @var null|EntityLoaderInterface
```

Comment:
`EntityLoaderInterface|null`

Just a detail, but this will keep code consistency. ;-)